File: test/fixtures/sample-code/undocumented.py
from dataclasses import dataclass


class ShoppingCart:
    def __init__(self):
        self.items = []

    def add_item(self, item):
        self.items.append(item)

    def get_total(self):
        return sum(item.price * item.quantity for item in self.items)


def calculate_total(items: list[float], tax: float) -> float:
    subtotal = sum(items)
    return subtotal * (1 + tax)


async def fetch_user_profile(user_id: str) -> dict:
    async with aiohttp.ClientSession() as session:
        async with session.get(f"/api/users/{user_id}") as response:
            return await response.json()


def _internal_helper():
    print("internal")

File: test/fixtures/sample-code/undocumented.ts
export function calculateTotal(items: number[], tax: number): number {
  const subtotal = items.reduce((sum, item) => sum + item, 0);
  return subtotal * (1 + tax);
}

export async function fetchUserProfile(userId: string): Promise<UserProfile> {
  const response = await fetch(`/api/users/${userId}`);
  return response.json();
}

export class ShoppingCart {
  private items: CartItem[] = [];

  addItem(item: CartItem): void {
    this.items.push(item);
  }

  getTotal(): number {
    return this.items.reduce((sum, item) => sum + item.price * item.quantity, 0);
  }
}

export interface UserProfile {
  id: string;
  name: string;
  email: string;
}

export type CartItem = {
  id: string;
  name: string;
  price: number;
  quantity: number;
};

function _internalHelper(): void {
  console.log('internal');
}
